ports/mimxrt/mpbthciport.c

```diff
@@ -47,6 +44,8 @@ uint8_t mp_bluetooth_hci_cmd_buf[4 + 256];
 STATIC mp_sched_node_t mp_bluetooth_hci_sched_node;
 STATIC soft_timer_entry_t mp_bluetooth_hci_soft_timer;
 
+extern void machine_uart_set_baudrate(mp_obj_t uart, uint32_t baudrate);
```

Comment:
This declaration should have the same type signature as its definition.  I suggest:
- moving this decl to `modmachine.h`
- use `struct _machine_uart_obj_t *` for the type of the first argument, so that struct doesn't need to be made public

Reply:
That was one of the reasons why I preferred the other variant. No declaration dependencies. But I'll do that.

Reply:
> use struct _machine_uart_obj_t * for the type of the first argument, so that struct doesn't need to be made public

That does not work. It creates a compile warning.

```
modmachine.h:53:39: error: 'struct _machine_uart_obj_t' declared inside parameter list will not be visible outside of this definition or declaration [-Werror]
   53 | void machine_uart_set_baudrate(struct _machine_uart_obj_t *uart, uint32_t baudrate);
```

Reply:
How about declaring the first argument of machine_uart_set_baudrate() as mp_obj_t and converting it in machine_uart_set_baudrate() to `machine_uart_t *` using the `MP_OBJ_TO_PTR()` macro. Like it's done in the functions for the methods.

Reply:
Updated accordingly.

Reply:
You could predeclare the struct in the header with `struct _machine_uart_obj_t;` on its own line.  But using `mp_obj_t` is also OK.